### ISSUE TITLE
Client crashes when crafting - NullPointerException when updateCollections called

### ISSUE THREAD

**Issue by EmotionalLove** [EmotionalLove/ReMinecraft#7]
```
[19:52:48] [main/INFO] [net.minecraft.client.gui.GuiNewChat]: [CHAT] ujustgotbaited killed themselves.
[19:52:49] [main/FATAL] [net.minecraft.client.Minecraft]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_181]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_181]
	at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1087) [bib.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:397) [bib.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:123) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_181]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_181]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_181]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_181]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
Caused by: java.lang.NullPointerException
	at net.minecraft.client.gui.recipebook.GuiRecipeBook.func_193003_g(GuiRecipeBook.java:172) ~[bno.class:?]
	at net.minecraft.client.gui.recipebook.GuiRecipeBook.func_193948_e(GuiRecipeBook.java:469) ~[bno.class:?]
	at net.minecraft.client.gui.inventory.GuiCrafting.func_192043_J_(SourceFile:138) ~[bmn.class:?]
	at net.minecraft.client.network.NetHandlerPlayClient.func_191980_a(NetHandlerPlayClient.java:1492) ~[brz.class:?]
	at net.minecraft.network.play.server.SPacketRecipeBook.func_148833_a(SourceFile:39) ~[js.class:?]
	at net.minecraft.network.play.server.SPacketRecipeBook.func_148833_a(SourceFile:12) ~[js.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(PacketThreadUtil.java:20) ~[hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_181]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_181]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
	... 9 more
```

I've tried blocking all recipe book related packets both clientbound and serverbound in RE:minecraft, the client still continues to crash (sometimes) upon crafting an item, like a torch.

**Comment by EmotionalLove** [comment]
May be fixed as of 4d450db2ecee9979d50e80f9f2dfa22e53cdb62c, about 30 minutes of playtime with constant crafting and no crashing

**Comment by EmotionalLove** [comment]
Closing an hour after testing.
